src/HTTP.php

```diff
@@ -495,6 +495,17 @@ public function parseRawBody()
                 if ($this->isJson() && is_null($this->parsed_body)) {
                     throw new ParsingException(json_last_error_msg(), json_last_error());
                 }
+            } else {
+                // e.g. Blocked by Cloudflare, and received a 403 page instead of a JSON response
+                $response = [
+                    "errorCode" => "security_block",
+                    "errorId" => null,
+                    "message" => "HTTP request blocked by security service. Cloudflare Ray ID: " . $this->getParsedHeaders()['cf-ray']
+                ];
```

Comment:
Is a Cloudflare security block the only reason that we may not get a JSON response? If not this message might be misleading. Maybe something like the below?

```suggestion
                $response = [
                    "errorCode" => "non_json_response",
                    "errorId" => null,
                    "message" => implode(" ", [
                        "HTTP expected JSON response. Received: ",
                        ($this->getContentType() ?: "unknown") . ".",
                        "Cloudflare Ray ID:",
                        isset($this->getParsedHeaders()['cf-ray']) ? $this->getParsedHeaders()['cf-ray'] : "not found"
                    ]);
                ];
```

The above has a test for multiple versions of php here: https://3v4l.org/oPfZY. Using implode is not necessary but thought it was easier to read than concatenation due to it having multiple expressions included.

Reply:
There's also a 4th property expected: `httpStatusCode`. This should be an int, and we can retrieve it from the cURL object.